### PR TITLE
[WFLY-5688] mod_cluster subsystem remains silent if connector set to undefined connector

### PR DESCRIPTION
**Pull request by iweiss**
Issue: https://issues.jboss.org/browse/WFLY-5688

### CODE REVIEW

**Comment by iweiss** [comment]
In my tests and also the integration tests, it seems this is enough. Am I missing something?

**Comment by pferraro** [comment]
This isn't quite right.  While the solution would indeed involve a capability reference, the connector value is meant to reference an undertow listener, not a socket binding.

**Comment by bstansberry** [comment]
Thanks for digging into this, Ingo.

This needs a Paul Ferraro-approved design of *all* the capabilities that will be installed by this subsystem, along with a PR for https://github.com/wildfly/wildfly-capabilities recording the necessary info for any of the caps in that design that we're ready to implement. It needs to be a real clear design of what caps there are, what they are meant to do and why they are named the way they are named. Keep in mind that the names of the capabilities of the server are exposed to end users.

In most cases it's not necessary to pass the capability into the add/remove handlers or do any special registration. See the SimpleResourceDefinition.Parameters class and the SimpleResourceDefinition constructor that takes it. Capabilities can be set in the Parameters class and then the kernel will take care of the registration. This stuff has evolved over the last 18 months, so there's code and perhaps docs that lag a bit behind the best practice way of doing it.

**Comment by iweiss** [comment]
@pferraro Hmmmm, so it's different from what I recall from EAP 6. In that case I think this PR, https://github.com/wildfly/wildfly/pull/8879, is more aligned with this, is it not?

@bstansberry Thanks for the instructions. I was reading through https://docs.jboss.org/author/display/WFLY10/Working+with+WildFly+Capabilities and was waiting for an approval before submitting a PR :-)

**Comment by bstansberry** [comment]
@iweiss I've updated that doc to better reflect how to do this now.

**Comment by iweiss** [comment]
@bstansberry I reworked this PR based on your and @pferraro's feedbacks.

The tests will fail because the only way I got the capability to get registered was using
```
private ModClusterConfigAdd() {
    super(ModClusterConfigResourceDefinition.ATTRIBUTES);
}
```

Which, I learned, will require a transformer to be in place. I'm working on the transformer in the meantime. I just wanted to make sure the code stays the up-to-date while I work on it.

File: mod_cluster/extension/src/main/java/org/wildfly/extension/mod_cluster/ModClusterConfigAdd.java [hunk 1]
```diff
@@ -35,6 +35,7 @@
     static final ModClusterConfigAdd INSTANCE = new ModClusterConfigAdd();
 
     private ModClusterConfigAdd() {
+        super(ModClusterConfigResourceDefinition.ATTRIBUTES);
```
Comment (bstansberry) [inline mod_cluster/extension/src/main/java/org/wildfly/extension/mod_cluster/ModClusterConfigAdd.java]: If you do this (which you should; it's best practice) then the populateModel impl below should be changed. It loops through the attributes in ModClusterConfigResourceDefinition.ATTRIBUTES. That can be replaced with a call to super.populateModel(....).

I'm not sure why this class treats ModClusterConfigResourceDefinition.SIMPLE_LOAD_PROVIDER differently.

**Comment by bstansberry** [comment]
@iweiss quick tip re: test failures. See org.jboss.as.subsystem.test.AdditionalInitialization and the use of its withCapabilities(...) and register[Service]Capabilities(...) methods in full. Those methods are about getting fake, placeholdres capabilities registered so tests can pass.

**Comment by iweiss** [comment]
I am very sorry for the spam guys. I'm still working on this and I mistakenly synchronised the branch.

File: mod_cluster/extension/src/main/java/org/wildfly/extension/mod_cluster/ModClusterConfigAdd.java (hunk 1 above)
Comment (ctomc) [inline mod_cluster/extension/src/main/java/org/wildfly/extension/mod_cluster/ModClusterConfigAdd.java]: Yeah that one is strange, when I looked trough history it just looks some cruft from the past.

**Comment by iweiss** [comment]
@bstansberry, @pferraro I think that's it gents. Local tests worked fine. Thanks for the hand @ctomc

**Comment by kabir** [comment]
@iweiss what is the status of this now?

**Comment by iweiss** [comment]
@kabir I need to discuss these failing tests and get back to it

**Comment by iweiss** [comment]
I just rebased this into the latest master and ran `./integration-tests.sh -DallTests` without any issues. I'm amending the commit and will push it soon. Hope it will pass the tests on CI.

**Comment by iweiss** [comment]
@kabir It seems the commit hash wasn't found. Do you oppose to me closing this PR and re-opening it?

**Comment by ctomc** [comment]
@iweiss @kabir there was temp issues on CI, which was hopefully fixed.

**Comment by iweiss** [comment]
I don't know why Windows is failing but I believe the Linux test fails probably due to missing transformers. @ctomc mentioned that transformers for some subsystems will be wrote in bulk due to Elytron. Would that be the case here as well?

**Comment by kabir** [comment]
Closing as per @pferraro 's feedback

**Pull request closed without merging**